### PR TITLE
Improve assertion error for hasSameElementsAs(emptyList())

### LINKED ISSUE #2294: nondescript assertion error for hasSameElementsAs(emptyList())
#### Summary
Compared to other asserts, `hasSameElementsAs` has only a brief and generic assertion error given an empty expected collection.

#### Example

```java
    List<String> actual = List.of("element");
    Assertions.assertThat(actual).as("helpful message").hasSameElementsAs(Collections.emptyList());
```
fails with
```
java.lang.AssertionError: actual is not empty
```

In comparision
```java
    List<String> actual = List.of("element");
    Assertions.assertThat(actual).as("helpful message").containsExactlyElementsOf(Collections.emptyList());
```
has a much richer message as I would expect from AssertJ:
```
java.lang.AssertionError: [helpful message] 
Expecting actual:
  ["element"]
to contain exactly (and in same order):
  []
but some elements were not expected:
  ["element"]
```

### PR DESCRIPTION
**Pull request by jhonnen**
Inlined commonCheckThatIterableAssertionSucceeds in assertContainsOnly and removed failIfEmptySinceActualIsNotEmpty call because expectedValues is not a subsequence.

#### Check List:
* Fixes #2294
* Unit tests : YES
* Javadoc with a code example (on API only) : NA
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

Following the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

### CODE REVIEW

File: src/test/java/org/assertj/core/api/iterable/IterableAssert_hasSameElementsAs_Test.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import static org.mockito.Mockito.verify;
 
 /**
- * Tests for <code>{@link org.assertj.core.api.AbstractIterableAssert#containsOnlyElementsOf(Iterable)} </code>.
+ * Tests for <code>{@link org.assertj.core.api.AbstractIterableAssert#hasSameElementsAs(Iterable)} </code>.
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/iterable/IterableAssert_hasSameElementsAs_Test.java:22]: good catch

File: src/main/java/org/assertj/core/internal/Iterables.java [hunk 2]
```diff
@@ -380,7 +380,9 @@ private void iterablesRemove(Iterable<?> actual, Object value) {
    */
   public void assertContainsOnly(AssertionInfo info, Iterable<?> actual, Object[] expectedValues) {
     final List<?> actualAsList = newArrayList(actual);
-    if (commonCheckThatIterableAssertionSucceeds(info, actualAsList, expectedValues)) return;
+    checkNotNullIterables(info, actualAsList, expectedValues);
+    // if both actual and values are empty, then assertion passes.
+    if (actualAsList.isEmpty() && expectedValues.length == 0) return;
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Iterables.java:385]: @jhonnen how does that address the issue?
`commonCheckThatIterableAssertionSucceeds` was doing the same thing with:
```
if (!actual.iterator().hasNext() && sequence.length == 0) return true;
```
You would need to add a test showing the improved behavior.

**Review by joel-costigliola — COMMENTED** [review]
@jhonnen PR commented!

File: src/main/java/org/assertj/core/internal/Iterables.java (hunk 2 above)
Comment (jhonnen) [reply src/main/java/org/assertj/core/internal/Iterables.java:385]: The issue was not caused by this check, but by `failIfEmptySinceActualIsNotEmpty(sequence)` from `commonCheckThatIterableAssertionSucceeds`.

**Comment by joel-costigliola** [comment]
Integrated thanks @jhonnen!

**Pull request closed without merging**